### PR TITLE
Fixes & improvements to Subtitle Chooser

### PR DESCRIPTION
**Pull request by svobs**
- [x] I have read [CONTRIBUTING.md](https://github.com/iina/iina/blob/develop/CONTRIBUTING.md)
- [x] This implements/fixes issue #4057.

---

**Description:**
Like #4825 but makes additional changes:
- The message `osd.sub_found`, previously set to `%d subtitles found. Downloading…`, was not exactly appropriate as the heading for the list of subtitles to download. It looks like this was done because it was used both for the subtitle list & for the actual "downloading" message. So I changed the existing `osd.sub_found` message to remove the ` Downloading…` part, added a new `osd.sub_downloading` message whose English version is `Downloading %d subtitles`, and changed the "downloading" message to use this key instead.
- Improve text size of sub-heading and table items slightly.
- Change table style & border so it looks like [this mock-up](https://github.com/iina/iina/issues/4057#issuecomment-1949629135) from @Iiii-I-I-I. Also implemented some other changes to displayed attributes based on said mock-up:
- Remove time from upload date (seems rare that subtitles are uploaded more than once a day).
- Remove rating (seems to be 0.0 most of the time?) - am open to bringing this back.
- Add release year in parentheses.
- Add up to 2 fraction digits to `fps` attribute's displayed value.

### CODE REVIEW

**Comment by svobs** [comment]
Squashed + rebased to fix merge errors + improved commit msg.

File: iina/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -83,7 +83,8 @@
 "osd.find_online_sub" = "Finding online subtitles…";
 "osd.find_online_sub.source" = "from";
 "osd.sub_not_found" = "No subtitles found";
-"osd.sub_found" = "%d subtitles found. Downloading…";
+"osd.sub_found" = "%d subtitles found.";
+"osd.sub_downloading" = "Downloading %d subtitles";
```
Comment (lhc70000) [inline iina/Base.lproj/Localizable.strings:92]: Please also add this to en.lproj.

File: iina/Extensions.swift [hunk 2]
```diff
@@ -290,6 +299,12 @@ extension FloatingPoint {
       return self
     }
   }
+
+  /// Formats as String, rounding the number to 2 digits after the decimal
+  var stringMaxFrac2: String {
```
Comment (lhc70000) [inline iina/Extensions.swift]: I suggest using a name like `stringWithMaxFractionDigit2` for better readability.

File: iina/OSDMessage.swift [hunk 3]
```diff
@@ -69,6 +69,7 @@ enum OSDMessage {
 
   case startFindingSub(String)  // sub source
   case foundSub(Int)
+  case downloadingSub(Int, String)  // download count, ssub source
```
Comment (lhc70000) [inline iina/OSDMessage.swift]: sub

**Review by lhc70000 — CHANGES_REQUESTED** [review]

File: iina/Extensions.swift (hunk 2 above)
Comment (svobs) [reply iina/Extensions.swift]: Ah yeah. For utility functions like this which might be commonly used, I worry about them getting too long. But, just semantics. Done

File: iina/OSDMessage.swift (hunk 3 above)
Comment (svobs) [reply iina/OSDMessage.swift]: Thanks

**Comment by svobs** [comment]
Rebased + pushed an update with all the suggested changes and more detailed commit message.

**Comment by lhc70000** [comment]
Merged via 1f87c39.

**Comment by low-batt** [comment]
This was included in IINA 1.4.0.

**Pull request closed without merging**